hikari/channels.py

```diff
@@ -717,6 +717,17 @@ class GuildChannel(PartialChannel):
     If no parent category is set for the channel, this will be `builtins.None`.
     """
 
+    @property
+    def mention(self) -> str:
```

Comment:
I have a feeling this might be implemented in the future, so we can add it to PartialChannel. It not being clickable on desktop is probably a bug, since it resolves

Reply:
Eh I'd lean in the side of assuming that it not being clickable on desktop is intended, they previously had the same behaviour in place for voice channels until around when public stages were added.
You also get the same behaviour for other channel types like guild categories for more obvious reasons and if it was a bug all of this behaviour would've been "solved" years ago since its really well known behaviour.

Reply:
Also out of curiosity why do you think it could be implemented in the future

Reply:
They had the same situation with voice channels back in the day. They got rendered with the icon but weren't clickable.

Also, random thought, having it in this class would also break when trying to mention a category?

Reply:
> They had the same situation with voice channels back in the day. They got rendered with the icon but weren't clickable.

Bit of a difference between a guild channel type they didn't allow mentioning until they brought in a new sub type of it which encouraged advertising it and a channel type which they don't allow mentioning cause its not a guild channel though.

Also my guess for why voice channel mentions weren't allowed isn't that it was a bug but rather that they'd previously thought voice channel mentions would be more annoying than they are helpful since they'd jump you straight into a possibly live voice chat and weren't bothered to bring in a way to differentiate between voice and text channel mentions since there'd be no real return on investment there.

Reply:
If they start allowing for DM channel mentions in the future this can simply be moved, nothing will break if they do that so this isn't exactly a case of preserving future compatibility. But suggesting behaviour which isn't possibly (dm channel mentions) might end up just leading to confusion and issues being made about it.

(Avoiding putting this method on GuildCategory would also be better if we want to avoid suggesting behaviour which doesn't exist but this'd lead to a lot of duplication and risks making mentioning guild channels using this method while following static typing more annoying than it's worth since they'd have to narrow down the scope of the variable further from GuildChannel)